src/components/vr-scatter-plot/vr-scatter-plot.component.ts

```diff
@@ -1,32 +1,95 @@
-import { Component, ElementRef, Input, OnChanges, OnDestroy, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
 import { Scatterplot } from '../../d3/scatterplot.d3';
-import 'aframe';
-
-
+import { PreferenceService } from '../../services/preference/preference.service';
+import { DataService } from '../../services/data/data.service';
+import { Meta } from '../../datasets/metas/types';
+import { TimeSeriesQueryOptions } from '../../datasets/queries/time-series.query';
+import { LineChartDatum} from '../line-chart/line-chart.component';
+import { LineChartMeta } from '../../datasets/metas/line-chart.meta';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { DAY } from '../../utils/timeUnits';
+import { map, takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-vr-scatter-plot',
   templateUrl: './vr-scatter-plot.component.html'
 })
-export class VRScatterPlotComponent implements OnInit, OnChanges, OnDestroy{
+export class VRScatterPlotComponent<T extends Meta>implements OnInit, OnChanges, OnDestroy{
+  @Input() endDate = new Date();
+  @Input() startDate = new Date(this.endDate.getTime() - 30 * DAY);
+  @Input() meta: Meta;
+  @Input() meta2: LineChartMeta;
   private vrScatterPlot: Scatterplot;
-  private shape: string;
-  private color: string;
+  componentMetas: Meta[];
+  queryOptions$ = new BehaviorSubject<TimeSeriesQueryOptions>({
+    range: [this.startDate, this.endDate],
+  });
+  datum$ = new BehaviorSubject<LineChartDatum>({
+    label: '',
+    points: [],
+  });
+  private destroy$ = new Subject();
 
   constructor(
+    private preferenceService: PreferenceService
   ) {
-    this.vrScatterPlot = new Scatterplot('a-sphere');
+    this.preferenceService = preferenceService;
   }
+
   ngOnInit() {
-   this.vrScatterPlot.init(document.querySelector('a-scene'), [10, 20, 30, 40, 50, 60]);
+    this.vrScatterPlot = new Scatterplot('a-sphere');
+    const dataService = new DataService(this.preferenceService);
+    dataService.dataset$
+    .pipe(takeUntil(this.destroy$))
+    .subscribe(dataset => {
+      // componentMetas is initialized to different dataset metas - will help funnel dataset
+      this.componentMetas = dataset.metas;
+      // dataset.metas[0].type = 'tabbed' and dataset.metas[1] = 'line'
+      this.meta = dataset.metas[0];
+      if (this.isTabbed()) {
+        this.meta2 = (this.meta as any).metas[0];
+        console.log(this.meta2);
```

Comment:
Can remove.

Reply:
@mfitz3 do you just mean the `console` printing statement?

Reply:
Yes